src/main/java/org/openjdk/jextract/impl/Utils.java

```diff
@@ -227,15 +230,19 @@ public static Class<?> carrierFor(Type.Primitive p) {
         };
     }
 
-    public static Class<?> valueLayoutCarrierFor(Type t) {
-        if (t instanceof Delegated delegated && delegated.kind() == Delegated.Kind.POINTER) {
-            return AddressLayout.class;
-        } else if (t instanceof Type.Primitive p) {
-            Class<?> clazz = carrierFor(p);
-            return CARRIERS_TO_LAYOUT_CARRIERS.get(clazz);
-        } else {
-            throw new UnsupportedOperationException(t.toString());
-        }
+    public static Class<?> layoutCarrierFor(Type t) {
```

Comment:
I have generalized this, so that we emit the layout carrier type for any jextract type